### PR TITLE
[AKS] support premium tier and add supportPlan

### PR DESCRIPTION
**Pull request by haitch**
**Related command**
az aks create
az aks update

**Description**
AKS start to offering LongTermSupport (LTS) for Kubernetes via Premium tier from 1.27, this CLI work is to complete the client experience.

**Testing Guide**
- command to create new aks cluster with LTS
```az aks create -g haitao -n ltsncus --kubernetes-support-plan AKSLongTermSupport -k 1.27.1 --tier premium -l southcentralus```

- command to onboard LTS feature for existing cluster
```az aks update -g haitao -n ltsncus --kubernetes-support-plan AKSLongTermSupport --tier premium```

- command to off-board LTS feature.
```az aks update -g haitao -n ltsncus --kubernetes-support-plan KubernetesOfficial```

**History Notes**

[AKS] `az aks create`: Add new parameter `--k8s-support-plan` to support LTS onboarding, also add new tier enum `premium`
[AKS] `az aks update`: Support to enable/disable LTS via new parameter `--k8s-support-plan`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:4709]:
```suggestion
        if self.mc and hasattr(self.mc, "support_plan") and self.mc.support_plan is not None:
```

should fix hybrid test

File: src/azure-cli/azure/cli/command_modules/acs/_format.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_format.py:153]: remove the leading and trailing space in the dict

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:432]: A little bit confusing, maybe `Choose from "KubernetesOfficial" or "AKSLongTermSupport", with ...`

**Review by FumingZhang — COMMENTED** [review]
Maybe add a new test for the new feature?

**Comment by haitch** [comment]
@FumingZhang need your help on the last test **test_aks_create_and_update_with_http_proxy_config**, it never works for me.

**Comment by FumingZhang** [comment]
The recording from [artifact](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=76037423&view=artifacts&pathAsName=false&type=publishedArtifacts) should works fine for test_aks_create_and_update_with_http_proxy_config

**Comment by FumingZhang** [comment]
And the [other one](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=76038728&view=artifacts&pathAsName=false&type=publishedArtifacts) works fine for test_aks_nodepool_snapshot

**Comment by haitch** [comment]
> Maybe add a new test for the new feature?

I will add that later via separate PR.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by zhoxing-ms** [comment]
Could you please add some test cases for this PR changes?

**Comment by zhoxing-ms** [comment]
> History Notes
> [AKS] az aks create: Support premium tier and LTS onboarding
[AKS] az aks update: Support to enable/disable LTS

Could you describe the `History Notes` more specifically? For example, mention the relevant parameters?

**Comment by haitch** [comment]
> Could you please add some test cases for this PR changes?

this would be blocked on server side pending some backend feature complete, we will add test once  the server side feature finish and unblock create of LTS cluster.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
